python/langsmith/evaluation/_runner.py

```diff
@@ -1655,15 +1672,19 @@ def _get_run(r: rt.RunTree) -> None:
 
 
 def _resolve_data(
-    data: DATA_T, *, client: langsmith.Client
+    data: DATA_T,
+    *,
+    client: langsmith.Client,
 ) -> Iterable[schemas.Example]:
     """Return the examples for the given dataset."""
+    # TODO: Find a smarter way of determining whether
+    # to get attachments (don't just default to true)
     if isinstance(data, str):
-        return client.list_examples(dataset_name=data)
+        return client.list_examples(dataset_name=data, get_attachments=True)
```

Comment:
how expensive can getting attachments be? should we make `get_attachments` an arg on resolve_data and turn `self.examples` into a method instead of a property that takes `get_attachments` as well? and then in with _predict and _score we pass get_attachments into self.examples(get_attachments)